javascript/index.js

```diff
@@ -24,17 +26,95 @@
+  }, (error) => console.log(error));
+}, (error) => console.log(error));
+
 
 // Iteration 2 - using promises
 // ...
 
+obtainInstruction('steak', 0)
+  .then( (step0) => {
+    document.querySelector("#steak").innerHTML += `<li>${step0}</li>`
+    return obtainInstruction('steak', 1)
+  })
+  .then( (step1) => {
+    document.querySelector("#steak").innerHTML += `<li>${step1}</li>`
+    return obtainInstruction('steak', 2)
+  })
+  .then( (step2) => {
+    document.querySelector("#steak").innerHTML += `<li>${step2}</li>`
+    return obtainInstruction('steak', 3)
+  })
+  .then( (step3) => {
+    document.querySelector("#steak").innerHTML += `<li>${step3}</li>`
+    return obtainInstruction('steak', 4)
+  })
+  .then( (step4) => {
+    document.querySelector("#steak").innerHTML += `<li>${step4}</li>`
+    return obtainInstruction('steak', 5)
+  })
+  .then( (step5) => {
+    document.querySelector("#steak").innerHTML += `<li>${step5}</li>`
+    return obtainInstruction('steak', 6)
+  })
+  .then( (step6) => {
+    document.querySelector("#steak").innerHTML += `<li>${step6}</li>`
+    document.querySelector("#steak").innerHTML += `<li>${`Stake is ready!`}</li>`
+    document.querySelector("#steakImg").removeAttribute(`hidden`)
+  })
+  .catch((err) => console.log(err));
+
 // Iteration 3 using async/await
 // ...
-
+async function makeBroccoli() {
+  try {
+    let broc0 = await obtainInstruction('broccoli', 0);
+    document.querySelector("#broccoli").innerHTML += `<li>${broc0}</li>`
+    let broc1 = await obtainInstruction('broccoli', 1);
+    document.querySelector("#broccoli").innerHTML += `<li>${broc1}</li>`
+    let broc2 = await obtainInstruction('broccoli', 2);
+    document.querySelector("#broccoli").innerHTML += `<li>${broc2}</li>`
+    let broc3 = await obtainInstruction('broccoli', 3);
+    document.querySelector("#broccoli").innerHTML += `<li>${broc3}</li>`
+    let broc4 = await obtainInstruction('broccoli', 4);
+    document.querySelector("#broccoli").innerHTML += `<li>${broc4}</li>`
+    let broc5 = await obtainInstruction('broccoli', 5);
+    document.querySelector("#broccoli").innerHTML += `<li>${broc5}</li>`
+    let broc6 = await obtainInstruction('broccoli', 6);
+    document.querySelector("#broccoli").innerHTML += `<li>${broc6}</li>`
+    document.querySelector("#broccoli").innerHTML += `<li>${`Broccoli is ready!`}</li>`
+    document.querySelector("#broccoliImg").removeAttribute("hidden");
+  } catch (error) {
+    console.log(error);
+  }
+}
+makeBroccoli();
 // Bonus 2 - Promise all
-// ...
\ No newline at end of file
+// ...
+
+//Promise.all([steak, mashedPotatoes, brusselsSprouts, broccoli])
+Promise.all([brusselsSprouts])
+.then((brusselsSprouts) => console.log(brusselsSprouts))
+.catch((err) => console.log(err));
+document.querySelector("#brusselsSprouts").innerHTML += `<li>${`Brussels sprouts are ready!`}</li>`
+document.querySelector("#brusselsSproutsImg").removeAttribute("hidden");
```

Comment:
Here, inside the Promise.all() we need to pass all the promises that we want to resolve, meaning that it would look something like this:
```
Promise.all([obtainInstruction("brusselsSprouts", 0),
  obtainInstruction("brusselsSprouts", 1),
  obtainInstruction("brusselsSprouts", 2),
  obtainInstruction("brusselsSprouts", 3),
  obtainInstruction("brusselsSprouts", 4),
  obtainInstruction("brusselsSprouts", 5),
  obtainInstruction("brusselsSprouts", 6),
  obtainInstruction("brusselsSprouts", 7),])
```

after this, on the ".then()" we would resolve all of those promises by looping over them:
```
Promise.all([
  obtainInstruction("brusselsSprouts", 0),
  obtainInstruction("brusselsSprouts", 1),
  obtainInstruction("brusselsSprouts", 2),
  obtainInstruction("brusselsSprouts", 3),
  obtainInstruction("brusselsSprouts", 4),
  obtainInstruction("brusselsSprouts", 5),
  obtainInstruction("brusselsSprouts", 6),
  obtainInstruction("brusselsSprouts", 7),
])
  .then((values) => {
    values.forEach((brussels) => {
      document.querySelector(
        "#brusselsSprouts"
      ).innerHTML += `<li>${brussels}</li>`;
    });
    document.querySelector("#brusselsSproutsImg").removeAttribute("hidden");
    document.querySelector(
      "#brusselsSprouts"
    ).innerHTML += `<li>Brussels sprouts are ready!</li>`;
  })
  .catch((error) => console.log(error));
```